ompi/mca/pml/ucx/pml_ucx_datatype.c

```diff
@@ -171,22 +153,67 @@ int mca_pml_ucx_datatype_attr_del_fn(ompi_datatype_t* datatype, int keyval,
     return OMPI_SUCCESS;
 }
 
+__opal_attribute_always_inline__
+static inline int mca_pml_ucx_datatype_is_contig(ompi_datatype_t *datatype)
+{
+    ptrdiff_t lb;
+
+    ompi_datatype_type_lb(datatype, &lb);
+
+    return (datatype->super.flags & OPAL_DATATYPE_FLAG_CONTIGUOUS) &&
+           (datatype->super.flags & OPAL_DATATYPE_FLAG_NO_GAPS) &&
+           (lb == 0);
+}
+
 #ifdef HAVE_UCP_REQUEST_PARAM_T
-__opal_attribute_always_inline__ static
-pml_ucx_datatype_t *mca_pml_ucx_init_nbx_datatype(ucp_datatype_t ucp_datatype,
-                                                  size_t size,
-                                                  int is_contig_pow2)
+/* NBX wrappers over request completion functions */
+/* it is compiled into single jump */
```

Comment:
maybe use inlining instead? e.g define those functions in pml_ucx_request.c and their prototypes in pml_ucx_request.h

Reply:
ok, moved into request.c file